lib/ukschedcoop/Config.uk

```diff
@@ -1,5 +1,14 @@
-config LIBUKSCHEDCOOP
+menuconfig LIBUKSCHEDCOOP
 	bool "ukschedcoop: Cooperative Round-Robin scheduler"
 	default y
 	select LIBNOLIBC if !HAVE_LIBC
 	select LIBUKSCHED
+
+if LIBUKSCHEDCOOP
+
+config LIBUKSCHEDCOOP_STATS
```

Comment:
Yes, it's purpose is to select `LIBUKSCHED_STATS`. That is because `libuksched` can't know at Kconfig level which library provides the implementation, so the path needs to be `LIBUKSCHEDCOOP_STATS` -> `LIBUKSCHED_STATS`. You will also notice that there is no option to enable stats in `libuksched`.

Reply:
Okay, understand. I would prefer if we design this similar to `ukalloc`, for simplicity. I would not provide an option to enable stats on each schedulers individually, I would enable them given what is configured in `lib/uksched`. In order to simplify instrumentation of the actual scheduler code we should provide helpers for counting events: https://github.com/unikraft/unikraft/blob/staging/lib/ukalloc/include/uk/alloc_impl.h#L161 Those helpers are either compiling in the code for counting or compiling it out, depending on how `uksched` is configured.

What I like is that any scheduler that doesn't support stats would keep the counters at `0` automatically, because the instrumentation would never be called. Although we have this effect already with the current code base.